app/src/main/java/org/kiwix/kiwixmobile/webserver/KiwixServer.kt

```diff
@@ -19,26 +19,37 @@
 package org.kiwix.kiwixmobile.webserver
 
 import android.util.Log
-import org.kiwix.kiwixlib.JNIKiwixException
-import org.kiwix.kiwixlib.JNIKiwixServer
-import org.kiwix.kiwixlib.Library
+import org.kiwix.libkiwix.Book
+import org.kiwix.libkiwix.JNIKiwixException
+import org.kiwix.libkiwix.Library
+import org.kiwix.libkiwix.Server
+import org.kiwix.libzim.Archive
 import javax.inject.Inject
 
 private const val TAG = "KiwixServer"
 
-class KiwixServer @Inject constructor(private val jniKiwixServer: JNIKiwixServer) {
+// jniKiwixServer is a server running on a existing library.
+// We must keep the library alive (keep a reference to it) to not delete the library the server
+// is working on.
```

Comment:
Here `Library` holds the books(Zim files) that we host on the `Server` (Kiwix Server) so for hosting books on the server we need to keep the library object otherwise it will crash the server because of library object is not alive.

Full conversation https://github.com/kiwix/java-libkiwix/issues/51

Reply:
Done